.github/workflows/test_s3.yml

```diff
@@ -0,0 +1,44 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  s3:
+    name: Test S3 functionality
+    runs-on: ubuntu-latest
+    env:
+      MINIO_ROOT_USER: admin
+      MINIO_ROOT_PASSWORD: password
+      MINIO_URL: http://localhost:9000
+      MINIO_ALIAS: myminio
+      MINIO_ACCESS_KEY: acquire
+      MINIO_SECRET_KEY: 12345678
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Install minio and mcli
+        run: |
+          apt update && apt install -y tmux wget
+          wget https://dl.min.io/server/minio/release/linux-amd64/minio -O /usr/local/bin/minio
+          wget https://dl.min.io/client/mc/release/linux-amd64/mc -O /usr/local/bin/mcli
+          chmod +x /usr/local/bin/minio
+          chmod +x /usr/local/bin/mcli
+
+      - name: Start minio in tmux
+        run: |
+          tmux new -d -s minio
+          tmux send-keys -t minio "MINIO_ROOT_USER=$MINIO_ROOT_USER MINIO_ROOT_PASSWORD=$MINIO_ROOT_PASSWORD minio server /tmp/minio --console-address :9001" Enter
+          sleep 5
```

Comment:
Why is it necessary to put the terminal to sleep? Also, how did we decide on 5 seconds?

Reply:
We can take this part out for now, but we'll need to give the server some time to spin up before we start exercising it.